### PR TITLE
Add Resource type ACM(AWS Certificate Manager)

### PR DESCRIPTION
**Pull request by inokappa**
Hi, Koyama san

I added AWS Certificate Manager Resource type.

### exist

```ruby
describe acm('example.com') do
  it { should exist }
end
```

### be_issued

```ruby
describe acm('example.com') do
  it { should be_issued }
end
```

### advanced

```ruby
describe acm('example.com') do
  its(:type) { should eq 'AMAZON_ISSUED' }
end
```

or

```ruby
describe acm('example.com') do
  its('type') { should eq 'AMAZON_ISSUED' }
end
```

Please confirm. :pray:

Regards.

Yohei.

### CODE REVIEW

File: doc/_resource_types/acm.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+```ruby
+describe acm('example.com') do
+  it { should be_issued }
+end
+```
+
+### advanced
+
+`acm` can use `Aws::ACM::Types::CertificateDetail` resource (see http://docs.aws.amazon.com/sdkforruby/api/Aws/ACM/Types/CertificateDetail.html).
```
Comment (k1LoW) [inline doc/_resource_types/acm.md]: This is not `advanced` 

##  What is advanced?

use `aws_resource` like ec2 type.

see https://speakerdeck.com/k1low/inside-of-awspec?slide=27

File: lib/awspec/type/acm.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  class Acm < Base
+    def resource_via_client
+      @resource_via_client ||= find_certificate(@display_name)
+    end
+
+    def id
+      @id = resource_via_client.certificate_arn if resource_via_client
+    end
+
+    def issued?
```
Comment (k1LoW) [inline lib/awspec/type/acm.rb]: Could you use `define_method` like https://github.com/k1LoW/awspec/blob/master/lib/awspec/type/ebs.rb#L23 ?

( see http://docs.aws.amazon.com/sdkforruby/api/Aws/ACM/Client.html#describe_certificate-instance_method and check `resp.certificate.status` )

File: lib/awspec/generator/doc/type/acm.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+module Awspec::Generator
+  module Doc
+    module Type
+      class Acm < Base
+        def initialize
+          super
+          @type_name = 'Acm'
+          @type = Awspec::Type::Acm.new('example.com')
+          @ret = @type.resource_via_client
+          @matchers = []
```
Comment (k1LoW) [inline lib/awspec/generator/doc/type/acm.rb]: Could you fix like  https://github.com/k1LoW/awspec/blob/master/lib/awspec/generator/doc/type/ebs.rb#L10 ?

File: lib/awspec/generator/doc/type/acm.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+          @ignore_matchers = []
```
Comment (k1LoW) [inline lib/awspec/generator/doc/type/acm.rb]: Could you fix like https://github.com/k1LoW/awspec/blob/master/lib/awspec/generator/doc/type/ebs.rb#L13 ?

**Comment by k1LoW** [comment]
Hi @inokappa !! Thank you for your active works!!
Please check my comment. 👍

**Comment by inokappa** [comment]
Hi @k1LoW !!!

Thank you for your advice!!!

I updated it with reference to your advice. :ok_man: 
Please confirm.

Regards.

Yohei

**Comment by k1LoW** [comment]
@inokappa LGTM!!!

**Comment by k1LoW** [comment]
Released as v0.69.0.

**Pull request merged**